apps/rush/src/actions/VersionAction.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { EOL } from 'os';
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import * as semver from 'semver';
+
+import {
+  CommandLineFlagParameter,
+  CommandLineStringParameter
+} from '@microsoft/ts-command-line';
+import {
+  IPackageJson,
+  IChangeInfo,
+  ChangeType,
+  RushConfigurationProject
+} from '@microsoft/rush-lib';
+
+import RushCommandLineParser from './RushCommandLineParser';
+import GitPolicy from '../utilities/GitPolicy';
+import { BaseRushAction } from './BaseRushAction';
+import { VersionManager } from '../utilities/VersionManager';
+import { Git } from '../utilities/Git';
+import ChangelogGenerator from '../utilities/ChangelogGenerator';
+
+export default class VersionAction extends BaseRushAction {
+  private _parser: RushCommandLineParser;
+  private _ensureVersionPolicy: CommandLineFlagParameter;
+  private _bumpVersion: CommandLineFlagParameter;
+  private _versionPolicy: CommandLineStringParameter;
+  private _bypassPolicy: CommandLineFlagParameter;
+  private _targetBranch: CommandLineStringParameter;
+  private _versionManager: VersionManager;
+
+  constructor(parser: RushCommandLineParser) {
+    super({
+      actionVerb: 'version',
+      summary: 'Manage package versions in the repo.',
+      documentation: 'use this "rush version" command to ensure version policies and bump versions.'
+    });
+    this._parser = parser;
+  }
+
+  protected onDefineParameters(): void {
+    this._targetBranch = this.defineStringParameter({
+      parameterLongName: '--target-branch',
+      parameterShortName: '-b',
+      key: 'BRANCH',
+      description:
+      'If this flag is specified, changes will be committed and merged into the target branch.'
+    });
+    this._ensureVersionPolicy = this.defineFlagParameter({
+      parameterLongName: '--ensure-version-policy',
+      parameterShortName: '-e',
+      description: 'Updates package versions if needed to satisfy version policies.'
+    });
+    this._bumpVersion = this.defineFlagParameter({
+      parameterLongName: '--bump',
+      description: 'Bumps package version based on version policies.'
+    });
+    this._bypassPolicy = this.defineFlagParameter({
+      parameterLongName: '--bypass-policy',
+      description: 'Overrides "gitPolicy" enforcement (use honorably!)'
+    });
+    this._versionPolicy = this.defineStringParameter({
+      parameterLongName: '--version-policy',
+      parameterShortName: '-p',
+      description: 'The name of the version policy'
+    });
+  }
+
+  protected run(): void {
+    if (!this._bypassPolicy.value) {
+      if (!GitPolicy.check(this.rushConfiguration)) {
+        process.exit(1);
+        return;
+      }
+    }
+    console.log(`Starting "rush version" ${EOL}`);
```

Comment:
Not in this PR.

---
In reply to: [125351378](https://github.com/Microsoft/web-build-tools/pull/259#discussion_r125351378) [](ancestors = 125351378)